src/parachain/issue.ts

```diff
@@ -176,24 +170,19 @@ export class DefaultIssueAPI implements IssueAPI {
             throw new Error("cannot request without setting account");
         }
 
-        let vault: VaultExt;
-        if (vaultId) {
-            vault = await this.vaultsAPI.get(vaultId);
-        } else {
+        if (!vaultId) {
             vaultId = await this.vaultsAPI.selectRandomVaultIssue(amountSat);
-            vault = await this.vaultsAPI.get(vaultId);
         }
         const griefingCollateralPlanck = await this.getGriefingCollateralInPlanck(amountSat.toString());
-        const requestIssueTx = this.api.tx.issue.requestIssue(amountSat, vault.id, griefingCollateralPlanck);
+        const requestIssueTx = this.api.tx.issue.requestIssue(amountSat, vaultId, griefingCollateralPlanck);
         const result = await sendLoggedTx(requestIssueTx, this.account, this.api);
         if (!this.isRequestSuccessful(result.events)) {
             Promise.reject("Request failed");
         }
 
         const id = this.getIssueIdFromEvents(result.events);
         const issueRequest = await this.getRequestById(id);
```

Comment:
I checked the UI again, can you return the entire issue request here please? Because we are getting the id and then we are fetching the request right again from the parachain. So we can safe one `await` call.